src/web/collectors/CollectorFilter.jsx

```diff
@@ -0,0 +1,188 @@
+        return 'Unknown';
+      case 2:
+        return 'Failing';
+      default:
+        return 'Unknown';
+    }
+  },
+  _transform(collectors) {
+    return collectors.map((collector) => {
+      const transformedCollector = {
+        id: collector.id,
+        name: collector.node_id,
+        tags: [''],
+        operating_system: '',
+        status: 'Unknown',
+      };
+      if (collector.node_details) {
+        transformedCollector.tags = collector.node_details.tags;
+        transformedCollector.operating_system = collector.node_details.operating_system;
+      }
+      if (collector.node_details.status) {
+        transformedCollector.status = this._getStatusText(collector.node_details.status.status);
+      }
+      return transformedCollector;
+    });
+  },
+  filterData() {
+    if (typeof this.props.filterData === 'function') {
+      return this.props.filterData(this.props.data);
+    }
+
+    const transformedData = this._transform(this.props.data);
+    const filteredData = transformedData.filter((datum) => {
+      return this._matchFilters(datum) && this._matchStringSearch(datum);
+    });
+
+    const mappedData = filteredData.map((datum) => {
+      return this.props.data.find((collector) => {
+        return collector.id === datum.id;
+      });
+    });
+
+    this.props.onDataFiltered(mappedData);
+  },
+  render() {
+    const filters = this.state.filters.map((filter) => {
+      return (
+        <li key={`li-${filter}`}>
+          <span className="pill label label-default">
+            {this.props.filterBy}: {filter}
+            <a className="tag-remove" data-target={filter} onClick={this._onFilterRemoved} />
+          </span>
+        </li>
+      );
+    });
+
+    let suggestions;
+
+    if (this.props.filterSuggestionAccessor) {
+      suggestions = this.props.filterSuggestions.map(filterSuggestion => filterSuggestion[this.props.filterSuggestionAccessor].toLocaleLowerCase());
+    } else if (this.props.filterSuggestions.length > 0){
+      suggestions = this.props.filterSuggestions.map(filterSuggestion => filterSuggestion.toLocaleLowerCase());
+    } else {
+      let filterByValues = [];
+      this._transform(this.props.data).forEach((collector) => {
+        if (!collector[this.props.filterBy]) {
+          return;
+        }
+        if (Array.isArray(collector[this.props.filterBy])) {
+          filterByValues = filterByValues.concat(collector[this.props.filterBy]);
+        } else {
+          filterByValues = filterByValues.concat([collector[this.props.filterBy]]);
+        }
+      });
+      suggestions = filterByValues.filter((value, index, self) => self.indexOf(value) === index);
+    }
+
+    suggestions.filter(filterSuggestion => !this.state.filters.includes(filterSuggestion));
+
+    return (
+      <div className="filter">
+        <form className="form-inline" onSubmit={this._onSearchTextChanged} style={{ display: 'inline' }}>
+          <TypeAheadInput ref="typeAheadInput"
```

Comment:
Same problem I described about string refs:
https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-string-refs.md

The easiest way to fix it is to replace the ref with something like: `ref={(c) => { this.__typeAheadInput = c; }}`, and then use `this._typeAheadInput` when accessing the component.